ddtrace/tracer.py

```diff
@@ -169,24 +193,34 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
                 span_type=span_type,
             )
 
-            span.set_tag(system.PID, getpid())
-
-            #    http://pypi.datadoghq.com/trace/docs/#distributed-tracing
-            parent_trace_id, parent_span_id = context._get_parent_span_ids()
-            if parent_trace_id:
-                span.trace_id = parent_trace_id
-
-            if parent_span_id:
-                span.parent_id = parent_span_id
-
-            self.sampler.sample(span)
+            span.sampled = self.sampler.sample(span)
+            if span.sampled:
+                # When doing client sampling in the client, keep the sample rate so that we can
+                # scale up statistics in the next steps of the pipeline.
+                if isinstance(self.sampler, RateSampler):
+                    span.set_metric(SAMPLE_RATE_METRIC_KEY, self.sampler.sample_rate)
```

Comment:
`set_metric()` is an hot-topic. We even have a discussion to remove it because it's confusing (it doesn't set a metric, it sets a tag/meta).

Reply:
Yes. That's for compatibility with what we are currently doing (SAMPLE_RATE_METRIC_KEY always was a metric).

We are likely to change it in the future.